test/integration/integration_test.cc

```diff
@@ -141,6 +141,97 @@ TEST_P(IntegrationTest, OverlyLongHeaders) {
 
 TEST_P(IntegrationTest, UpstreamProtocolError) { testUpstreamProtocolError(); }
 
+TEST_P(IntegrationTest, WebSocketConnectionDownstreamDisconnect) {
+  // WebSocket upgrade, send some data and disconnect downstream
+  IntegrationTcpClientPtr tcp_client;
+  FakeRawConnectionPtr fake_upstream_connection;
+  const std::string upgrade_req_str = "GET /websocket/test HTTP/1.1\r\nHost: host\r\nConnection: "
+                                      "Upgrade\r\nUpgrade: websocket\r\n\r\n";
+  const std::string upgrade_resp_str =
+      "HTTP/1.1 101 Switching Protocols\r\nConnection: Upgrade\r\nUpgrade: websocket\r\n\r\n";
+  executeActions({
+      [&]() -> void { tcp_client = makeTcpConnection(lookupPort("http")); },
+      // Send websocket upgrade request
+      [&]() -> void { tcp_client->write(upgrade_req_str); },
+      [&]() -> void { fake_upstream_connection = fake_upstreams_[0]->waitForRawConnection(); },
+      [&]() -> void { fake_upstream_connection->waitForData(190); },
+      // Accept websocket upgrade request
+      [&]() -> void { fake_upstream_connection->write(upgrade_resp_str); },
+      [&]() -> void { tcp_client->waitForData(upgrade_resp_str); },
+      // Standard TCP proxy semantics post upgrade
+      [&]() -> void { tcp_client->write("hello"); },
+      [&]() -> void { fake_upstream_connection->waitForData(195); },
+      [&]() -> void { fake_upstream_connection->write("world"); },
+      [&]() -> void { tcp_client->waitForData(upgrade_resp_str + "world"); },
+      [&]() -> void { tcp_client->write("bye!"); },
+      // downstream disconnect
+      [&]() -> void { tcp_client->close(); },
+      [&]() -> void { fake_upstream_connection->waitForData(199); },
+      [&]() -> void { fake_upstream_connection->waitForDisconnect(); },
+  });
+}
+
+#if 0
+  // HELP: Eventhough the initial part of this test is identical to the other websocket
```

Comment:
Not sure if this helps any but last time I hit this I literally copy-pasted the old test and replaced my new one line by line until I found the difference.
Good luck either way!